scalding-serialization-macros/src/main/scala/com/twitter/scalding/serialization/macros/impl/ordered_serialization/TreeOrderedBuf.scala

```diff
@@ -102,19 +103,24 @@ object TreeOrderedBuf {
         q"""
 
       override def dynamicSize($element: $typeName): Option[Int] = {
-        val $tempLen = payloadLength($element) match {
-          case _root_.com.twitter.scalding.serialization.macros.impl.ordered_serialization.runtime_helpers.NoLengthCalculation => None
-          case _root_.com.twitter.scalding.serialization.macros.impl.ordered_serialization.runtime_helpers.ConstLen(l) => Some(l)
-          case _root_.com.twitter.scalding.serialization.macros.impl.ordered_serialization.runtime_helpers.DynamicLen(l) => Some(l)
-        }
-        (if ($tempLen.isDefined) {
-          // Avoid a closure here while we are geeking out
-          val innerLen = $tempLen.get
-          val $lensLen = posVarIntSize(innerLen)
-          Some(innerLen + $lensLen)
-       } else None): Option[Int]
+        if(skipLenCalc) None else {
+          val $tempLen = payloadLength($element) match {
+            case _root_.com.twitter.scalding.serialization.macros.impl.ordered_serialization.runtime_helpers.NoLengthCalculation =>
+              failedLengthCalc()
+              None
+            case _root_.com.twitter.scalding.serialization.macros.impl.ordered_serialization.runtime_helpers.ConstLen(l) => Some(l)
+            case _root_.com.twitter.scalding.serialization.macros.impl.ordered_serialization.runtime_helpers.DynamicLen(l) => Some(l)
+          }
+          (if ($tempLen.isDefined) {
+            // Avoid a closure here while we are geeking out
+            val innerLen = $tempLen.get
+            val $lensLen = posVarIntSize(innerLen)
+            Some(innerLen + $lensLen)
+         } else None): Option[Int]
```

Comment:
just realized the macros are not hygienic when it comes to Option/Some/None. Maybe put an:

`import _root_.Predef.{Option, Some, None}` on line 235?

testing for clean imports is such a pain with macros.